modules/scripts/startup-script/examples/install_ansible.sh

```diff
@@ -165,14 +168,17 @@ main() {
 	fi
 	pip_version=$(${python_path} -m pip --version | sed -nr 's/^pip ([0-9]+\.[0-9]+).*$/\1/p')
 	pip_major_version=$(echo "${pip_version}" | cut -d '.' -f 1)
-	if [ "${pip_major_version}" -lt "${REQ_PIP_MINOR_VERSION}" ]; then
+	if [ "${pip_major_version}" -lt "${REQ_PIP_MAJOR_VERSION}" ]; then
 		${python_path} -m pip install --upgrade pip
 	fi
 
+	ghpc_venv="/usr/local/ghpc-venv"
 	# Create pip virtual environment for HPC Toolkit
-	${python_path} -m pip install virtualenv
-	${python_path} -m virtualenv /usr/local/ghpc-venv
+	${python_path} -m venv ${ghpc_venv}
```

Comment:
Data on startup time. Things to note:
- This is using the hpc image. It must have nfs client already installed as it is just exiting fast
- Huge improvement for ansible install time 👍 
- Virtualenv setup got slower. On the `after` trial this is including the install of `setuptools` and `wheel` which are being downloaded. It seems like virtualenv got these for free (same version). 3s is not much but if it is all the same I would say stick with virtualenv. 

  | Before | After
-- | -- | --
  | Total | Total
Python Install | 0:00:09 | 0:00:10
Pip Install | 0:00:03 | 0:00:02
Virtualenv | 0:00:03 | 0:00:06
Install Ansible | 0:02:44 | 0:00:41
install nfs | 0:00:02 | 0:00:02
mount | 0:00:07 | 0:00:07

Reply:
I am following the "leave the campsite better than you found it" rule. For me that means:

- upgrade system `pip` to something that is modern enough to install packages respecting
- otherwise, do not install any system-level packages (even `virtualenv` itself)

So, in my view, the onus is upon `virtualenv` to demonstrate its value. You have identified one difference: `virtualenv` will install [seed packages](https://virtualenv.pypa.io/en/latest/user_guide.html#wheels) for pip, setuptools, and wheel. It's a good idea to have recent versions of these. By contrast, `venv` uses `ensurepip` to install a copy of pip that is included with the Python distribution itself (i.e. the original system copy).

So.... ultimately the choice is to:

1. download `virtualenv` and its 3 embedded wheels for `pip`, `setuptools`, and `wheel`
2. install `virtualenv` system-wide
3. create a virtual environment with pre-staged versions of `pip`, `setuptools`, and `wheel` as chosen by `virtualenv`

OR

1. Use the built-in `venv` and install versions of `pip`, `setuptools`, `wheel` as we see fit.

Between following the campsite rule and having freedom to choose, I prefer the latter approach.